src/main/java/picocli/AutoComplete.java

```diff
@@ -750,7 +785,7 @@ private static String generatePositionalParamsCases(List<PositionalParamSpec> po
             int max = param.index().max();
             if (param.completionCandidates() != null) {
                 buff.append(format("%s    %s (( currIndex >= %d && currIndex <= %d )); then\n", indent, ifOrElif, min, max));
-                buff.append(format("%s      positionals=$( compgen -W \"$%s_pos_param_args\" -- \"%s\" )\n", indent, paramName, currWord));
+                buff.append(format("%s      positionals=$( compReplyArray \"${%s_pos_param_args[@]}\" )\n", indent, paramName, currWord));
```

Comment:
We recently [discovered](https://github.com/remkop/picocli/issues/2053) that the format string has only 2 `%s` variables, but we specify 3 parameters (`indent`, `paramName`, `currWord`)...

This means that the last parameter (`currWord`) is not used.
Is that correct? Can the `currWord` parameter be removed?

Reply:
Yeah, it can be removed. Sorry for missing that

Reply:
fixed in #2175